lib/Microservice.js

```diff
@@ -0,0 +1,141 @@
+const { promisify } = require('util');
+const loopback = require('loopback');
+const boot = require('loopback-boot');
+
+
+/**
+ * Basic Microservice class wrapping a loopback application.
+ *
+ * @type {module.Microservice}
+ */
+
+module.exports = class Microservice {
+
+    constructor(app, bootOptions = {}) {
+        const defaultAppDir = './app';
+        const defaultConfigDir = `${defaultAppDir}/config`;
+
+        this.app = app;
+        this.server = null;
+        this._bootOptions = Object.assign({
+            appRootDir: defaultAppDir,
+            appConfigRootDir: defaultConfigDir,
+            componentRootDir: defaultConfigDir,
+            dsRootDir: defaultConfigDir,
+            env: 'dev',
+            middlewareRootDir: defaultConfigDir,
+            modelsRootDir: defaultConfigDir,
+        }, bootOptions);
+    }
+
+    /**
+     * Stop the webserver of the microservice.
+     *
+     * @returns {Promise.<Microservice>}
+     */
+    async stop() {
+
+        if (!this.server || !this.server.listening) return Promise.reject(new Error('Microservice not running'));
```

Comment:
you may just throw an error here

Reply:
the error message is not very descriptive, how about «Cannot stop the microservice, it is not running»